src/main/scala/strawman/collection/View.scala

```diff
@@ -290,22 +287,107 @@ trait ArrayLike[+A] extends Any {
   def apply(i: Int): A
 }
 
+trait SeqView[+A] extends View[A] with SeqOps[A, SeqView, SeqView[A]] { self =>
+  override def view: SeqView[A] = this
+
+  override def toSeq = immutable.IndexedSeq.from(this)
+
+  override def iterableFactory: IterableFactoryLike[SeqView] = SeqView
+
+  override protected[this] def fromSpecificIterable(coll: Iterable[A]): SeqView[A] = fromIterable(coll)
+
+  override protected[this] def newSpecificBuilder(): Builder[A, SeqView[A]] =
+    IndexedSeq.newBuilder().mapResult(_.view)
+
+  override def toString = "SeqView(?)"
```

Comment:
We could implement it once in `View` like so:

~~~ scala
override def toString = s"$className(?)"
~~~